modules/reactor/src/meshgenerators/ReactorGeometryMeshBuilderBase.C

```diff
@@ -254,3 +268,50 @@ ReactorGeometryMeshBuilderBase::print2dMetadataToConsole(const std::string metad
   for (const auto & row : metadata_value)
     _console << "    " << Moose::stringify(row) << std::endl;
 }
+
+void
+ReactorGeometryMeshBuilderBase::addDepletionId(MeshBase & input_mesh,
+                                               const MooseEnum & option,
+                                               const DepletionIDGenerationLevel generation_level,
+                                               const bool extrude)
+{
+  // prepare set of extra elem ids for depletion ID generation
+  std::vector<ExtraElementIDName> id_names;
+  if (generation_level == DepletionIDGenerationLevel::Core)
+  {
+    if (option == "pin")
+      id_names = {"assembly_id", "pin_id"};
+    else if (option == "pin_type")
+      id_names = {"assembly_id", "pin_type_id"};
+    else if (option == "assembly")
+      id_names = {"assembly_id"};
+    else if (option == "assembly_type")
+      id_names = {"assembly_type_id"};
+  }
+  else if (generation_level == DepletionIDGenerationLevel::Assembly)
+  {
+    if (option == "pin")
+      id_names = {"pin_id"};
+    else if (option == "pin_type")
+      id_names = {"pin_type_id"};
+    else
+      paramError("depletion_id_type",
+                 "'pin' or 'pin_type' is not allowed in depletion ID generation at assembly level");
+  }
+  else if (generation_level == DepletionIDGenerationLevel::Pin)
+    mooseError("Depletion ID generation is not supported at pin level yet in RGMB");
+  if (extrude)
+    id_names.push_back("plane_id");
+  id_names.push_back("region_id");
+  // prepare list of blocks
```

Comment:
do we not have a getter for the list of blocks

Reply:
also why are you block-restricting to all blocks? the routine call after will be faster if you pass an empty set (meaning not block-restricted)